script/DeployLeveragedEthMainnet.s.sol

```diff
@@ -45,23 +53,69 @@ contract DeployScript is CREATE3Script {
 
         scWeth = new scWETH(scWethParams);
 
+        weth.deposit{value: 0.01 ether}(); // wrap 0.01 ETH into WETH
+        deposit(scWeth, weth, 0.01 ether); // 0.01 WETH
+
+        // transfer DEFAULT_ADMIN_ROLE to multisig
+        scWeth.grantRole(DEFAULT_ADMIN_ROLE, C.MULTISIG);
+        scWeth.revokeRole(DEFAULT_ADMIN_ROLE, deployerAddress);
+
         scUSDC.ConstructorParams memory scUsdcParams = scUSDC.ConstructorParams({
-            admin: address(this),
+            admin: deployerAddress,
             keeper: keeper,
             scWETH: scWeth,
-            usdc: ERC20(C.USDC),
+            usdc: usdc,
             weth: WETH(payable(C.WETH)),
             aavePool: IPool(C.AAVE_POOL),
             aavePoolDataProvider: IPoolDataProvider(C.AAVE_POOL_DATA_PROVIDER),
             aaveAUsdc: IAToken(C.AAVE_AUSDC_TOKEN),
             aaveVarDWeth: ERC20(C.AAVAAVE_VAR_DEBT_WETH_TOKEN),
-            uniswapSwapRouter: ISwapRouter(C.UNISWAP_V3_SWAP_ROUTER),
+            uniswapSwapRouter: uniswapRouter,
             chainlinkUsdcToEthPriceFeed: AggregatorV3Interface(C.CHAINLINK_USDC_ETH_PRICE_FEED),
             balancerVault: IVault(C.BALANCER_VAULT)
         });
 
         scUsdc = new scUSDC(scUsdcParams);
 
+        swapETHForUSDC(0.01 ether);
+        deposit(scUsdc, usdc, usdc.balanceOf(address(deployerAddress))); // 0.01 ether worth of USDC
+
+        // transfer DEFAULT_ADMIN_ROLE to multisig
+        scUsdc.grantRole(DEFAULT_ADMIN_ROLE, C.MULTISIG);
+        scUsdc.revokeRole(DEFAULT_ADMIN_ROLE, deployerAddress);
+
         vm.stopBroadcast();
     }
+
+    function deposit(sc4626 vault, ERC20 underlying, uint256 amount) internal {
+        underlying.approve(address(vault), amount);
+        vault.deposit(amount, deployerAddress);
+    }
+
+    function swapETHForUSDC(uint256 amount) internal {
+        weth.deposit{value: amount}();
+
+        weth.approve(address(uniswapRouter), amount);
+
+        address[] memory tokenPath = new address[](2);
+        tokenPath[0] = address(weth);
+        tokenPath[1] = address(usdc);
+
+        uint256[] memory amountsMin = new uint256[](2);
+        amountsMin[0] = amount;
+        amountsMin[1] = 0;
```

Comment:
this seems unnecessary, right?

Reply:
I mean you are creating two arrays just to pass the values you assign to the struct initialized below, just use the values directly

Reply:
so true :facepalm: removed